test/integration/src/test/java/io/pravega/test/integration/RestoreBackUpDataRecoveryTest.java

```diff
@@ -319,6 +345,37 @@ public void close() {
         }
     }
 
+
+    /**
+     * Creates a Pravega instance.
+     */
+    private static class PravegaRunner {
+        private final int containerCount;
+        private BookKeeperRunner bookKeeperRunner;
```

Comment:
This one is only set in the constructor. Make it final.

Reply:
It is set in one more place within the test to restart bk/zk.